packages/logging/src/log.js

```diff
@@ -591,14 +609,22 @@ Log.prototype.write = function(entry, options, callback) {
  *
  * @param {object} entry - An entry object.
  */
-Log.prototype.formatEntryForApi_ = function(entry) {
-  if (!(entry instanceof Entry)) {
-    entry = this.entry(entry);
-  }
+Log.prototype.decorateEntries_ = function(entries, callback) {
+  var self = this;
 
-  var formattedEntry = entry.toJSON();
-  formattedEntry.logName = this.formattedName_;
-  return formattedEntry;
+  async.map(entries, function(entry, callback) {
```

Comment:
Thanks for writing that up. I think let's just stick with the current way, then consider doing a `async.map` -> `Promise` sweep across the repo at a time in the future.